### PR TITLE
GH75: Template PackAsTool & nuspec->csproj

### PR DESCRIPTION
**Pull request by devlead**
This PR switches from using nuspec to csproj, uses same build.ps1 as frosting, adds build.sh and config.

As 38.3 doesn't exist switched back to 0.38.2 so template could be tested.

~~Now that I think of it this might remove the need for NuGet exe.~~

NuGet.exe dependency also removed for building frosting itself.

### CODE REVIEW

File: template/Cake.Frosting.Template.csproj [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <PackageType>Template</PackageType>
+    <PackageId>Cake.Frosting.Template</PackageId>
+    <Title>Cake.Frosting templates for the .NET SDK.</Title>
+    <Description>Cake.Frosting templates for the .NET SDK.</Description>
+    <Authors>Patrik Svensson, Mattias Karlsson, Gary Ewan Park, Alistair Chapman, Martin Björkström and contributors</Authors>
```
Comment (gep13) [inline template/Cake.Frosting.Template.csproj:8]: I think we are missing some names here...

File: template/Cake.Frosting.Template.csproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <Copyright>Copyright (c) .NET Foundation and contributors</Copyright>
+    <PackageTags>Cake;Script;Build</PackageTags>
+    <PackageIcon>cake-medium.png</PackageIcon>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <RepositoryType>git</RepositoryType>
+    <RepositoryUrl>https://github.com/cake-build/frosting</RepositoryUrl>
```
Comment (gep13) [inline template/Cake.Frosting.Template.csproj:14]: This should have .git at the end of it.

File: template/Cake.Frosting.Template.csproj (hunk 1 above)
Comment (devlead) [reply template/Cake.Frosting.Template.csproj:8]: Ah just copied
https://github.com/cake-build/frosting/blob/3f3c3cf4c849ff11d92d3374ea1bdbfe1ebf6287/src/Cake.Frosting/Cake.Frosting.csproj#L6

I'll create a Shared msbuild like we've got with Cake so all common properties in one place.

File: template/Cake.Frosting.Template.csproj (hunk 2 above)
Comment (devlead) [reply template/Cake.Frosting.Template.csproj:14]: Should this be corrected too then?
https://github.com/cake-build/cake/blob/daa3c4f26a5d54cbb8c0b65301eca098d2703f01/src/Shared.msbuild#L10

File: template/Cake.Frosting.Template.csproj (hunk 1 above)
Comment (gep13) [reply template/Cake.Frosting.Template.csproj:8]: That seems like a great idea!

File: template/Cake.Frosting.Template.csproj (hunk 2 above)
Comment (gep13) [reply template/Cake.Frosting.Template.csproj:14]: Yes, I would say so.  Having .git at the end of the URL is the recommendation that came from NuGet Team: https://twitter.com/adgrv/status/1127591787697061888

Comment (devlead) [reply template/Cake.Frosting.Template.csproj]: Corrected and moved to Shared.msbuild 
image

File: template/Cake.Frosting.Template.csproj (hunk 1 above)
Comment (devlead) [reply template/Cake.Frosting.Template.csproj]: Moved to Shared.msbuild 
image

File: src/Shared.msbuild [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<Project>
+  <PropertyGroup>
+    <VersionPrefix>0.1.0</VersionPrefix>
+    <PackageId>$(AssemblyName)</PackageId>
+    <Copyright>Copyright (c) .NET Foundation and contributors</Copyright>
+    <Authors>Patrik Svensson, Mattias Karlsson, Gary Ewan Park, Alistair Chapman, Martin Björkström, Dave Glick, Pascal Berger, Jérémie Desautels, Enrico Campidoglio and contributors</Authors>
+    <Company>Patrik Svensson, Mattias Karlsson, Gary Ewan Park, Alistair Chapman, Martin Björkström, Dave Glick, Pascal Berger, Jérémie Desautels, Enrico Campidoglio and contributors</Company>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <PackageIcon>cake-medium.png</PackageIcon>
+    <PackageIconUrl>https://raw.githubusercontent.com/cake-build/graphics/master/png/cake-medium.png</PackageIconUrl>
```
Comment (gep13) [inline src/Shared.msbuild:10]: @devlead I don’t think we need this one, right?

Comment (devlead) [reply src/Shared.msbuild:10]: Don't need, but it'll be used if feed/tool doesn't understand new format.

Comment (gep13) [reply src/Shared.msbuild:10]: Oh, it will be used as a fallback? I didn’t know that, I thought it was one, or the other!

**Comment by gep13** [comment]
@devlead your changes have been merged, thanks for your contribution 👍

**Pull request merged**